contracts/InterestRate/StableRate.sol

```diff
@@ -0,0 +1,240 @@
+     */
+    function updateStableRateModel(
+        uint256 _baseRatePerBlockForStable,
+        uint256 _multiplierPerBlock,
+        uint256 _multiplierPerBlockForStable,
+        uint256 _jumpMultiplierPerBlockForStable,
+        uint256 _kink,
+        uint256 _stableRatePremium,
+        uint256 _optimalStableLoanRate
+    ) external virtual {
+        require(msg.sender == owner, "StableRateModel: only owner may call this function.");
+
+        updateStableRateModelInternal(
+            _baseRatePerBlockForStable,
+            _multiplierPerBlock,
+            _multiplierPerBlockForStable,
+            _jumpMultiplierPerBlockForStable,
+            _kink,
+            _stableRatePremium,
+            _optimalStableLoanRate
+        );
+    }
+
+    /**
+     * @notice Calculates the utilization rate of the market: `borrows / (cash + borrows - reserves)`
+     * @param cash The amount of cash in the market
+     * @param borrows The amount of borrows in the market
+     * @param reserves The amount of reserves in the market
+     * @return The utilization rate as a mantissa between [0, BASE]
+     */
+    function utilizationRate(
+        uint256 cash,
+        uint256 borrows,
+        uint256 reserves
+    ) public pure returns (uint256) {
+        // Utilization rate is 0 when there are no borrows
+        if (borrows == 0) {
+            return 0;
+        }
+
+        return (borrows * BASE) / (cash + borrows - reserves);
+    }
+
+    /**
+     * @notice Calculates the stable loan rate of the market: `stableBorrows / totalBorrows`
+     * @param stableBorrows The amount of stable borrows in the market
+     * @param totalBorrows The amount of total borrows in the market
+     * @return The stable loan rate as a mantissa between [0, BASE]
+     */
+    function stableLoanRate(uint256 stableBorrows, uint256 totalBorrows) public pure returns (uint256) {
+        // Loan rate is 0 when there are no stable borrows
+        if (totalBorrows == 0) {
+            return 0;
+        }
+
+        return (stableBorrows * BASE) / totalBorrows;
+    }
+
+    /**
+     * @notice Calculates the current borrow rate per block, with the error code expected by the market
+     * @param cash The amount of cash in the market
+     * @param stableBorrows The amount of stable borrows in the market
+     * @param totalBorrows The amount of borrows in the market
+     * @param reserves The amount of reserves in the market
+     * @return The borrow rate percentage per block as a mantissa (scaled by BASE)
+     */
+    function getBorrowRate(
+        uint256 cash,
+        uint256 stableBorrows,
+        uint256 totalBorrows,
+        uint256 reserves
+    ) external view returns (uint256) {
+        uint256 util = utilizationRate(cash, totalBorrows, reserves);
+        uint256 loanRate = stableLoanRate(stableBorrows, totalBorrows);
+        uint256 excessLoanRate = calculateLoanRateDiff(loanRate);
+
+        if (util <= kink) {
```

Comment:
Are you using the formulas described in the Appendix of [this document](https://docs.google.com/document/d/1DxrzeuPb9N1uO4AmVmGjxFb7r7dNrg7_nY934RVq46A/edit#)? Just to confirm I'm checking the correct ones (because the formulas in the section "Interest Rate Model with Stable Interest Rate Option" are different)

Reply:
Yes, we are using formulas described in the Appendix of [this doc](https://docs.google.com/document/d/1DxrzeuPb9N1uO4AmVmGjxFb7r7dNrg7_nY934RVq46A/edit#)

Reply:
Ok. I was initially confused because I saw a comment from Lynn in that document saying we won't use those formulas

Reply:
Yes, you were right. I have refactored the calculations of the Stable rate borrowing.